### ISSUE TITLE
Unable to resolve transitive dependency (possible upstream POM error?)

### ISSUE THREAD

**Issue by mtomko** [alexarchambault/coursier#258]
I attempted to migrate several projects to coursier. Most worked swimmingly, but one was unable to resolve a dependency and reported:

```
[info] Updating gpp-lims-server:gpp-lims-server_2.11:1.0.0
Ignoring unparsed versions: Vector(${version})
Ignoring unparsed versions: Vector(${version})
Ignoring unparsed versions: Vector(${version})
Ignoring unparsed versions: Vector(${version})
Ignoring unparsed versions: Vector(${version})
Ignoring unparsed versions: Vector(${version})
Ignoring unparsed versions: Vector(${version})
Ignoring unparsed versions: Vector(${version})
Ignoring unparsed versions: Vector(${version})
Ignoring unparsed versions: Vector(${version})
Ignoring unparsed versions: Vector(${version})
Ignoring unparsed versions: Vector(${version})
Ignoring unparsed versions: Vector(${version})
Ignoring unparsed versions: Vector(${version})
Ignoring unparsed versions: Vector(${version})
Ignoring unparsed versions: Vector(${version})
[info] Resolved gpp-lims-server:gpp-lims-server_2.11:1.0.0 dependencies
[info] Updating gpp-lims-client:gpp-lims-client_sjs0.6_2.11:1.0.0
[info] Resolved gpp-lims-client:gpp-lims-client_sjs0.6_2.11:1.0.0 dependencies
[info] Fetching artifacts of gpp-lims-server:gpp-lims-server_2.11:1.0.0
[info] Fetched artifacts of gpp-lims-server:gpp-lims-server_2.11:1.0.0
[error] 1 not found
[error]   https://repo1.maven.org/maven2/org/apache/ws/commons/XmlSchema/SNAPSHOT/XmlSchema-SNAPSHOT.jar
[info] Fetching artifacts of gpp-lims-client:gpp-lims-client_sjs0.6_2.11:1.0.0
[info] Fetched artifacts of gpp-lims-client:gpp-lims-client_sjs0.6_2.11:1.0.0
[trace] Stack trace suppressed: run last server/*:update for the full output.
[error] (server/*:update) java.lang.Exception: Encountered 1 errors (see above messages)
[error] Total time: 2 s, completed May 25, 2016 10:45:15 AM
```

I tried to create a minimal repro case, since this project is somewhat large (it's a Play project) and is not open source. The dependency in question is coming in via a dependency on Atlassian's Crowd authentication framework, which we can't reasonably avoid here.

I was able to create a minimal build file that exhibits most of the same behavior, although it never prints the message about `Ignoring unparsed versions: Vector(${version})`. Here's the build.sbt:

``` scala
lazy val root = (project in file("."))
  .settings(
    name := "coursier-test",
    version := "1.0",
    scalaVersion := "2.11.8",
    libraryDependencies ++= Seq("org.codehaus.xfire" % "xfire-core" % "1.2.6")
  )
```

I used a `project/plugins.sbt` to include `addSbtPlugin("io.get-coursier" % "sbt-coursier" % "1.0.0-M12")` and `project/build.properties` to set the SBT version to `0.13.11`.

With this minimal project, I get:

```
> last *:update
[info] Fetching artifacts of coursier-test:coursier-test_2.11:1.0
[info] Fetched artifacts of coursier-test:coursier-test_2.11:1.0
[error] 1 not found
[error]   https://repo1.maven.org/maven2/org/apache/ws/commons/XmlSchema/SNAPSHOT/XmlSchema-SNAPSHOT.jar
java.lang.Exception: Encountered 1 errors (see above messages)
    at coursier.Tasks$$anonfun$updateTask$1.coursier$Tasks$$anonfun$$report$1(Tasks.scala:710)
    at coursier.Tasks$$anonfun$updateTask$1$$anonfun$apply$49.apply(Tasks.scala:748)
    at coursier.Tasks$$anonfun$updateTask$1$$anonfun$apply$49.apply(Tasks.scala:748)
    at scala.collection.mutable.MapLike$class.getOrElseUpdate(MapLike.scala:189)
    at scala.collection.mutable.AbstractMap.getOrElseUpdate(Map.scala:91)
    at coursier.Tasks$$anonfun$updateTask$1.apply(Tasks.scala:741)
    at coursier.Tasks$$anonfun$updateTask$1.apply(Tasks.scala:517)
    at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
    at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
    at sbt.std.Transform$$anon$4.work(System.scala:63)
    at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
    at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
    at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
    at sbt.Execute.work(Execute.scala:237)
    at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
    at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
    at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
    at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
[error] (*:update) java.lang.Exception: Encountered 1 errors (see above messages)
>
```

After looking around at other issues, I noticed #157 which looked similar. Following the path there, I was able to create an even more minimal repro case:

``` bash
% coursier fetch "org.codehaus.xfire":"xfire-core":"1.2.6"
  Error:
https://repo1.maven.org/maven2/org/apache/ws/commons/XmlSchema/SNAPSHOT/XmlSchema-SNAPSHOT.jar: NotFound(https://repo1.maven.org/maven2/org/apache/ws/commons/XmlSchema/SNAPSHOT/XmlSchema-SNAPSHOT.jar,Some(true))
```

However, unlike #157, disabling checksums does not help the matter:

``` bash
 % coursier fetch "org.codehaus.xfire":"xfire-core":"1.2.6" --checksum none
  Error:
https://repo1.maven.org/maven2/org/apache/ws/commons/XmlSchema/SNAPSHOT/XmlSchema-SNAPSHOT.jar: NotFound(https://repo1.maven.org/maven2/org/apache/ws/commons/XmlSchema/SNAPSHOT/XmlSchema-SNAPSHOT.jar,Some(true))
```

Like #157, the POM does not appear to include versions for dependencies.

http://central.maven.org/maven2/org/codehaus/xfire/xfire-core/1.2.6/xfire-core-1.2.6.pom

Perhaps that's a red herring, though.

**Comment by alexarchambault** [comment]
The `SNAPSHOT` version for `org.apache.ws.commons:XmlSchema` seems to originate from... the `SNAPSHOT` version in http://repo1.maven.org/maven2/org/apache/ws/commons/XmlSchema/1.1/XmlSchema-1.1.pom. That's a bug though. It should be easy to fix.

**Comment by alexarchambault** [comment]
https://github.com/alexarchambault/coursier/pull/259 should fix the issue with `org.apache.ws.commons:XmlSchema`.

The `Ignoring unparsed versions` messages aren't a good sign for your `gpp-lims-server:gpp-lims-server_2.11:1.0.0` though.

**Comment by mtomko** [comment]
I was wondering about those. I found the line of code in Coursier that emits that warning, but didn't get a chance to try and figure out what it meant. In practice, the system seems to work - maybe that's the ecosystem of JVM dependency management for you...

**Comment by FranklinChen** [comment]
I think I'm running into a related problem:

```
$ coursier fetch -r https://www.phon.ca/artifacts/libs-snapshot ca.phon:phontalk-core:2.0.0-SNAPSHOT
```

```
Error:
  ca.phon:phon-project:${ca.phon.version}:
    not found: /Users/chen/.ivy2/local/ca.phon/phon-project/${ca.phon.version}/ivys/ivy.xml
    not found: https://repo1.maven.org/maven2/ca/phon/phon-project/$%7Bca.phon.version%7D/phon-project-$%7Bca.phon.version%7D.pom
    not found: https://www.phon.ca/artifacts/libs-snapshot/ca/phon/phon-project/$%7Bca.phon.version%7D/phon-project-$%7Bca.phon.version%7D.pom
  ca.phon:phon-syllabifier:${ca.phon.version}:
    not found: /Users/chen/.ivy2/local/ca.phon/phon-syllabifier/${ca.phon.version}/ivys/ivy.xml
    not found: https://repo1.maven.org/maven2/ca/phon/phon-syllabifier/$%7Bca.phon.version%7D/phon-syllabifier-$%7Bca.phon.version%7D.pom
    not found: https://www.phon.ca/artifacts/libs-snapshot/ca/phon/phon-syllabifier/$%7Bca.phon.version%7D/phon-syllabifier-$%7Bca.phon.version%7D.pom
```

**Comment by alexarchambault** [comment]
@FranklinChen This is related to https://github.com/alexarchambault/coursier/issues/169 (namely, a parent POM uses snapshot versioning, and is not found, which provokes the errors you're seeing).

**Comment by alexarchambault** [comment]
@FranklinChen With https://github.com/coursier/coursier/pull/567 which should be included in `1.0.0-RC4`, the following works:
```
$ coursier fetch ca.phon:phontalk-core:2.0.0-SNAPSHOT -r https://www.phon.ca/artifacts/libs-release -r https://www.phon.ca/artifacts/libs-snapshot
```
